sandbox/testinputs/functions.hpp

```diff
@@ -0,0 +1,15 @@
+#ifndef FUNCTIONS_HPP
+#define FUNCTIONS_HPP
+
+#include <vector>
+
+template <typename T>
+inline T distribution(const std::vector<T> &in)
```

Comment:
This is the actual test function.  It will get compiled into two different object files, one with some optimization flags and another without.